Network/Haskoin/Node/Mempool.hs

```diff
@@ -154,6 +167,29 @@ processTxInv pid tids = do
         -- Publish the transaction download job with average priority
         sendManager $ PublishJob (JobDwnTxs toDwn) (ThisPeer pid) 5
 
+processGetTx :: (MonadIO m, MonadLogger m)
+             => PeerId
+             -> TxHash
+             -> StateT MempoolSession m ()
+processGetTx pid tid = do
+    $(logDebug) $ format $ unwords
+        [ "Requesting transaction", encodeTxHashLE tid, "from wallet." ]
+    modify $
+        \s -> s{ txPeerMap = insertWith (flip (++)) tid [pid] (txPeerMap s) }
+    sendWallet $ WalletGetTx tid
+
+processSendTx :: (Applicative m, MonadIO m, MonadLogger m)
```

Comment:
I have nothing against Applicative per se, but to stay consistent with the rest of the code in the node, I would suggest using liftM as we have that already with the Monad constraints.